### PR TITLE
Add CITATION.cff file and remove Zenodo

### PR DESCRIPTION
**Pull request by AntonyMilneQB**
## Description
Completing the good work of Zain from https://github.com/quantumblacklabs/kedro/pull/862. As per the discussions there, I've:
* removed all references to Zenodo
* put only team members on the list
* included the version number because it will be updated on release. I'll add a note to our release docs to make sure the date gets updated too.

Here's what it looks like:

image

## Checklist

- [x] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change and added my name to the list of supporting contributions in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [x] Added tests to cover my changes

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,4 +119,5 @@ Kedro has also won [Best Technical Tool or Framework for AI](https://awards.ai/t
 
 ## How can I cite Kedro?
 
-If you're an academic, Kedro can also help you, for example, as a tool to solve the problem of reproducible research. Find our citation reference on [Zenodo](https://zenodo.org/record/4336685).
+If you're an academic, Kedro can also help you, for example, as a tool to solve the problem of reproducible research. Use the "Cite this repository" button on [our repository](https://github.com/quantumblacklabs/kedro) to generate a citation from the [CITATION.cff file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#about-citation-filesat).
+
```
Comment (AntonyMilneQB) [inline README.md:123]:
```suggestion
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- family-names: Milne
+  given-names: Antony
+- family-names: Schwarzmann
+  given-names: Joel
+- family-names: Theisen
+  given-names: Merel
+- family-names: Wong
+  given-names: Susanna
+title: Kedro
+version: 0.17.5
```
Comment (MerelTheisenQB) [inline CITATION.cff:29]: We might want to add a note to the release checklist to also update the version and release data here.

**Review by MerelTheisenQB — APPROVED** [review]
LGTM 👍

**Review by yetudada — APPROVED** [review]
Happy for this! Thank you for pointing us in this direction @mzjp2!

File: CITATION.cff (hunk 2 above)
Comment (AntonyMilneQB) [reply CITATION.cff:29]: Will do 🙂

File: docs/source/12_faq/01_faq.md [hunk 3]
```diff
@@ -183,7 +183,7 @@ Our preferred Kedro-community channel for feedback is through [GitHub issues](ht
 
 ## How can I cite Kedro?
 
-If you're an academic, Kedro can also help you, for example, as a tool to solve the problem of reproducible research. Find our citation reference on [Zenodo](https://zenodo.org/record/4336685).
+If you're an academic, Kedro can also help you, for example, as a tool to solve the problem of reproducible research. Use the "Cite this repository" button on [our repository](https://github.com/quantumblacklabs/kedro) to generate a citation from the [CITATION.cff file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files#about-citation-filesat).
```
Comment (AntonyMilneQB) [inline docs/source/12_faq/01_faq.md:186]:
```suggestion
If you're an academic, Kedro can also help you, for example, as a tool to solve the problem of reproducible research. Use the "Cite this repository" button on [our repository](https://github.com/quantumblacklabs/kedro) to generate a citation from the [CITATION.cff file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files).
```

**Pull request merged**